moninag/tests/unittests/registration/test_view.py

```diff
@@ -0,0 +1,316 @@
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, expected)
+
+    def test_login(self):
+        """Ensure that login() method works properly"""
+
+        user_data = json.dumps({'email': "testmail@test.so",
+                                'password': "rootroot",
+                                })
+        url = reverse('login')
+        response = self.client.post(url, data=user_data, content_type='application/json')
+
+        self.assertEqual(response.status_code, 200)
+
+    def test_login_is_not_active(self):
+        """Ensure we get 401 response if user is not active"""
+
+        self.user.is_active = False
+        self.user.save()
+        user_data = json.dumps({'email': "testmail@test.so",
+                                'password': "rootroot",
+                                })
+
+        url = reverse('login')
+        response = self.client.post(url, data=user_data, content_type='application/json')
+
+        self.assertEqual(response.status_code, 401)
+
+    def test_login_invalid_email_password(self):
+        """Ensure we can't log in with invalid email/password"""
+
+        user_data = json.dumps({'email': "abyrvalg",
+                                'password': "glavryba",
+                                })
+
+        url = reverse('login')
+        response = self.client.post(url, data=user_data, content_type='application/json')
+
+        self.assertEqual(response.status_code, 403)
+
+    def test_login_method_GET(self):
+        """Ensure we get a right response when using !=POST method"""
+
+        url = reverse('login')
+        response = self.client.get(url)
+
+        self.assertEqual(response.status_code, 400)
+
+    def test_logout(self):
+        """Ensure we can successfully log out."""
+
+        self.client.login(username="testmail@test.so", password="rootroot")
+        urllogout = reverse('logout')
+        response_out = self.client.get(urllogout)
+
+        self.assertRedirects(response_out, '/auth/',
+                             status_code=302, target_status_code=200)
+
+    def test_register_user(self):
+        """Ensure that we can successfully register a user."""
+
+        reg_data = json.dumps({'email': 'testmail2@test.so',
+                               'firstName': 'TestName',
+                               'lastName': 'LastName',
+                               'password': 'isecretlyloveunittests'})
+
+        url = reverse('register_user')
+        response = self.client.post(url, data=reg_data, content_type='application/json')
+        expected_json = json.dumps({'error': {},
+                                    'message': 'Thank you for your time. The confirmation code has been sent to '
+                                    'your email. In order to confirm the registration, simply click on '
+                                    'the link given in it.',
+                                    'success': True})
```

Comment:
isn't it 88, not 99? The code will be ugly if I make it to 88. I've managed to shorten it t 99 symbols though.